listener.go

```diff
@@ -168,7 +167,7 @@ func setSubscriptionFilters(ctx context.Context, l *Listener) error {
 }
 
 // Listen waits for a message from the Service Bus Topic subscription
-func (l *Listener) Listen(ctx context.Context, handle Handle, topicName string, opts ...ListenerOption) error {
+func (l *Listener) Listen(ctx context.Context, handler handle.Handler, topicName string, opts ...ListenerOption) error {
```

Comment:
Confused why the user would pass in a handler here rather than a handle. Maybe I'll figure out still.

Reply:
Okay a handle is a handler becsuse you've implmented Do on it. 
This is confusing but not incorrect. (still not used to functions havign method recievers on them )

Reply:
This is the common handler pattern in go. :)